boot/freeldr/freeldr/ntldr/wlmemory.c

```diff
@@ -238,6 +239,60 @@ WinLdrSetupMemoryLayout(IN OUT PLOADER_PARAMETER_BLOCK LoaderBlock)
         return FALSE;
     }
 
+    // Always map acpi table if present
+    if (Extension->AcpiTable)
+    {
+        Status = MempSetupPaging((ULONG_PTR)Extension->AcpiTable >> MM_PAGE_SHIFT, 
+                                 ((ULONG_PTR)Extension->AcpiTableSize + MM_PAGE_SIZE) >> MM_PAGE_SHIFT,
+                                 FALSE);
+        if (!Status)
+        {
+            ERR("MempSetupPaging() failed\n");
+            return FALSE;
+        }
+
+        PACPISDTHeader AcpiTable = Extension->AcpiTable;
+        BOOLEAN is64BitsTable = _strnicmp(AcpiTable->Signature, "XSDT", 4) == 0;
+        ULONG Entries = (AcpiTable->Length - sizeof(ACPISDTHeader)) / (is64BitsTable ? 8 : 4);
+
+        if (is64BitsTable)
+        {
+            PXSDT_DESCRIPTOR Xsdt = (PXSDT_DESCRIPTOR)AcpiTable;
+
+            for (ULONG i = 0; i < Entries; i++)
```

Comment:
Which exact usecase is this PR trying to fix? Which system can it boot with your PR, that it couldn't boot before?

Ftr this type of decalring variables won't build with older compilers, which do require declaring variables at a blocks top.

Reply:
https://github.com/reactos/reactos/blob/master/ntoskrnl/po/power.c#L459-L462
This is required for George's Po work, UEFI, and is a thing needed for later versions of Windows along with SMBIOS

Reply:
Providing ACPI tables in a member field helps me understanding the anatomy of the hardware in terms of power management such as the presence of hibernation support, sleep states, if the hardware can handle modern sleep (aka AoAc), etc, of which during power management initialization `PoInitSystem` I can report the power properties of the hardware accordingly based on what the ACPI tables say. 

That would require some further helper function to decode the information though. But yes, this is needed.

Reply:
> Declaring variables at the top is taught as a bad programming practice

Not explaining the actual reason makes this assertion more suspicious.

Reply:
I disagree that declaring variables at the top is bad practice. Depending on the individual case it can both either increase or decrease readability / debuggability.
Our C standard is C99 and compilers that are so ancient, that they cannot handle it, cannot compile our code base anyway, so @JoachimHenze, please stop with this stupid argument. We have moved on and we are not going back.